expression/builtin_cast.go

```diff
@@ -249,7 +253,7 @@ func (b *castAsStringFunctionClass) getFunction(args []Expression, ctx context.C
 			sig = &builtinCastStringAsStringSig{bf}
 		}
 	}
-	return sig, errors.Trace(b.verifyArgs(args))
+	return sig.setSelf(sig), errors.Trace(b.verifyArgs(args))
```

Comment:
Why do we need to set self-value？
If we really need to do that，line 237 also need to be set.